desktop/core/src/desktop/js/parse/hplsqlStatementsParser.test.js

```diff
@@ -0,0 +1,97 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import hplsqlStatementsParser from './hplsqlStatementsParser';
+
+describe('hplsqlStatementsParser.js', () => {
+  const stringifySplitResult = function (result) {
+    let s = '[';
+    let first = true;
+    result.forEach(entry => {
+      s += first ? '{\n' : ', {\n';
+      s += "  statement: '" + entry.statement.replace(/\n/g, '\\n') + "',\n";
+      s +=
+        '  location: { first_line: ' +
+        entry.location.first_line +
+        ', first_column: ' +
+        entry.location.first_column +
+        ', last_line: ' +
+        entry.location.last_line +
+        ', last_column: ' +
+        entry.location.last_column +
+        ' }';
+      if (entry.firstToken) {
+        s += ",\n  firstToken: '" + entry.firstToken + "'";
+      }
+      if (entry.database) {
+        s += ",\n  database: '" + entry.database + "'";
+      }
+      s += '\n}';
+      first = false;
+    });
+    s += ']';
+    return s;
+  };
+
+  const testParser = function (input, expectedOutput) {
+    try {
+      expectedOutput.forEach(entry => {
+        entry.type = 'statement';
+      });
+      const result = hplsqlStatementsParser.parse(input);
+      const because =
+        '\n\nParser output: ' +
+        stringifySplitResult(result) +
+        '\nExpected output: ' +
+        stringifySplitResult(expectedOutput);
+
+      expect(result).toEqual(expectedOutput, because);
+    } catch (error) {
+      console.error(error);
+      console.warn(error.message);
+
+      fail('Got error');
+    }
+  };
+
+  it('should split "CREATE PROCEDURE greet(name STRING)\\nBEGIN\\n  PRINT \'Hello \' || name;END;" correctly', () => {
```

Comment:
You're missing a bunch of tests for comments etc. One solution is to extract those tests into something like:

```
export const sharedStatementParserTests(statementParser) => {
  it ('...') { ... }
  ...
}
```

and then add a line here, `sharedStatementParserTests(hplsqlStatementsParser);`

Same goes for the `testParser` and `stringifySplitResult` to get rid of the duplication.

Reply:
> > @JohanAhlen How ro enable this parser when hplsql mode is on?
> 
> I'd create a `utils.ts` next to the new parsers with an `export const getStatementsParser = (connector: Connector): sqlStatementsParser => { ... }` that returns the correct parser depending on some attribute of the connector.
> 
> Then search the codebase for `import sqlStatementsParser` and instead import and call the new function in all places where you can find it. i.e. change `sqlStatementsParser.parse(...)` to `getStatementsParser(connector).parse(...)`

Done, it is working now.
When connector is 'hive' and hplsql mode is on then hplsqlStatementsParser works otherwise sqlStatementsParser parser works.

Reply:
can't understand how to export all the examples?